client/extensions/woocommerce/app/promotions/promotion-form-type-header.js

```diff
@@ -25,21 +23,23 @@ const PromotionFormTypeCard = ( {
 		const data = { type: e.target.value };
 		if ( 'coupon' === data.type ) {
 			data.coupon = { code: '', discount_type: 'percent', amount: '' };
+			data.product = undefined;
+		}
+		if ( 'product_sale' === data.type ) {
+			data.product = {};
+			data.coupon = undefined;
 		}
 		editPromotion( siteId, promotion, data );
 	};
 
 	return (
-		<Card className="promotions__promotion-form-type">
-			<FormFieldset className="promotions__promotion-form-type-select">
-				<FormLabel>{ translate( 'Promotion type' ) }</FormLabel>
-				<FormSelect value={ promotionType } onChange={ onTypeSelect }>
-					<option value="" disabled>{ translate( 'Select promotion type…' ) }</option>
-					<option value="coupon">{ translate( 'Coupon' ) }</option>
-					<option value="product_sale">{ translate( 'Product Sale' ) }</option>
-				</FormSelect>
-			</FormFieldset>
-		</Card>
+		<SectionHeader label={ translate( 'Promotion type' ) } className="promotions__promotion-form-type-header">
+			<FormSelect value={ promotionType } onChange={ onTypeSelect }>
+				<option value="" disabled>{ translate( 'Select promotion type…' ) }</option>
+				<option value="coupon">{ translate( 'Coupon' ) }</option>
+				<option value="product_sale">{ translate( 'Product Sale' ) }</option>
```

Comment:
Might want to add context here for translators, since Sale could mean Purchase or could mean Offered at a Reduced Price.